needle/modules/configuration/mdm_assess.py

```diff
@@ -0,0 +1,131 @@
+            regex = '(^ *)|(= *\{.*)|(;)'
+            parsed.append('\n'.join(re.sub(regex, '', x) for x in element.split('\n')[:-1]))
+
+        return parsed
+
+    # Compare two config files
+    def compare(self, fConfig, fDesired):
+        config  = self.parseConfigData(fConfig)
+        desired = self.parseConfigData(fDesired)
+        misConfigs = 0
+
+        # Print output header
+        print "\n"+40*"-"
+        self.printer.notify("MDM Configuration Assessment")
+        print 40*"-"
+        
+        # Compare attribute
+        for i in range(len(config))[:-1]:
+            # Check for termplate/config attribute mismatch
+            if config[i].split("\n")[0] != desired[i].split("\n")[0]:
+                self.printer.warning("Mismatch found! Invalid Template.")
+                self.printet.debug("%s >> %s" % (onfig[i].split("\n")[0],desired[i].split("\n")[0]))
+                exit(1)
+
+            # Check templace compliance
+            if config[i] != desired[i]:
+                self.printer.error("[ BAD] " + config[i].split("\n")[0])
+                misConfigs += 1
+
+                # Print misconfiguration details Verbosely
+                if self.options["verbosity"] >= 2:
+                    cAttribs = config[i].split("\n")[1:]
+                    dAttribs = desired[i].split("\n")[1:]
+                    for i in range(len(cAttribs)):
+                        print "\b    ",; self.printer.verbose(cAttribs[i])
+                        if cAttribs[i] != dAttribs[i] and self.options["verbosity"] >= 3:
+                            print "\b    ",; self.printer.notify(dAttribs[i]  + " [RECOMMENDED]")
+            else:
+                if self.options["verbosity"] == 4:
+                    # Print non-isconfigured attributes
+                    self.printer.info( "[GOOD] " + config[i].split("\n")[0])
+
+        # Print output footer
+        print '\b'+40*'-'
+        self.printer.notify("%s/%s Misconfigurations" % (misConfigs, len(desired)))
+        print 40*'-'+"\n"
+
+    # ==================================================================================================================
+    # RUN
+    # ==================================================================================================================
+    def module_run(self):
+        # Find MDM config file locations
+        self.printer.verbose("Searching for Configuration file...")
+
+        arg = "/var/mobile/Library/ConfigurationProfiles/EffectiveUserSettings.plist"
+        cmd = '{bin} {arg}'.format(bin=self.device.DEVICE_TOOLS['FIND'], arg=arg)
+        config = self.device.remote_op.command_blocking(cmd)[0].strip()
+
+        if not config:
+            self.printer.error("No Configuration files found!")
+            return
+        
+        # Place template config within /tmp/ directory
+        template = "/tmp/" + self.options['template'].split('/')[-1]
+        self.device.push(self.options["template"], template)
```

Comment:
This is pushed to the device as it may require being parsed with plutil if it is not in json format and the user may not have plutil locally on their machine.